handler.go

```diff
@@ -358,6 +370,17 @@ func (s *streamIOInitializer) HandleResponse(body []byte) error {
 	return nil
 }
 
+// allRequests returns all the requests from the stream reader.
+func (s *streamIOInitializer) allRequests() [][]byte {
+	for !s.eofReached {
+		if _, err := s.NextRequest(); err != nil && err != io.EOF {
+			s.out.Fatalf("%v\n", err)
```

Comment:
Since allRequests was being invoked directly handleStreamRequest, it made sense to handle fatal directly as it has access to out.Fatal. Anyway, I have moved Fatal out of allRequest :-)